packages/tempus/src/Tempus_TimeStepControlStrategyComposite.hpp

```diff
@@ -62,26 +73,168 @@ class TimeStepControlStrategyComposite : virtual public TimeStepControlStrategy<
+  virtual Teuchos::RCP<const Teuchos::ParameterList> getValidParameters() const override
+  {
+    Teuchos::RCP<Teuchos::ParameterList> pl =
+      Teuchos::parameterList("Time Step Control Strategy");
+
+    std::stringstream sList;
+    for(std::size_t i = 0; i < strategies_.size(); ++i) {
+      sList << strategies_[i]->getStrategyType();
+      if (i < strategies_.size()-1) sList << ", ";
+    }
+
+    pl->set<std::string>("Strategy Type", this->getStrategyType(), "Integral Controller");
+
+    pl->set<std::string>("Strategy List", sList.str());
+
+    for(auto& s : strategies_)
+      pl->set(s->getName(), *s->getValidParameters());
+
+    return pl;
   }
 
 private:
+
   std::vector<Teuchos::RCP<TimeStepControlStrategy<Scalar > > > strategies_;
 
 };
+
+
+// Nonmember constructor - ParameterList
+// ------------------------------------------------------------------------
+template <class Scalar>
+Teuchos::RCP<TimeStepControlStrategyComposite<Scalar> >
+createTimeStepControlStrategyComposite(
+  Teuchos::RCP<Teuchos::ParameterList> const& pList,
+  std::string name = "Composite Strategy")
+{
+  using Teuchos::RCP;
+  using Teuchos::ParameterList;
+
+  std::vector<std::string> tscsList;
+
+  TEUCHOS_TEST_FOR_EXCEPTION(
+    pList->get<std::string>("Strategy Type", "Composite Strategy") !=
+      "Composite Strategy", std::logic_error,
+    "Error - Strategy Type != 'Composite Strategy'.  (='"
+    +pList->get<std::string>("Strategy Type")+"')\n");
+
+  // string tokenizer
+  tscsList.clear();
+  std::string str = pList->get<std::string>("Strategy List");
+  std::string delimiters(",");
+  const char* WhiteSpace = " \t\v\r\n";
+  // Skip delimiters at the beginning
+  std::string::size_type lastPos = str.find_first_not_of(delimiters, 0);
+  // Find the first delimiter
+  std::string::size_type pos     = str.find_first_of(delimiters, lastPos);
+  while ((pos != std::string::npos) || (lastPos != std::string::npos)) {
+    // Found a token, add it to the vector
+    std::string token = str.substr(lastPos,pos-lastPos);
+
+    std::size_t start = token.find_first_not_of(WhiteSpace);
+    std::size_t end = token.find_last_not_of(WhiteSpace);
+    token = (start == end ? std::string() : token.substr(start, end-start+1));
+
+    tscsList.push_back(token);
+    if(pos==std::string::npos) break;
+
+    lastPos = str.find_first_not_of(delimiters, pos); // Skip delimiters
+    pos = str.find_first_of(delimiters, lastPos);     // Find next delimiter
+  }
+
+  auto tscsc = Teuchos::rcp(new TimeStepControlStrategyComposite<Scalar>());
+
+  // For each sublist name tokenized, add the TSCS
+  for( auto tscsName: tscsList){
+    RCP<ParameterList> pl =
+      Teuchos::rcp(new ParameterList(pList->sublist(tscsName)));
+
+    auto strategyType = pl->get<std::string>("Strategy Type");
+    if (strategyType == "Constant") {
+      tscsc->addStrategy(
+        createTimeStepControlStrategyConstant<Scalar>(pl, tscsName));
```

Comment:
I'm getting a compilation error with a missing `#include "Tempus_TimeStepControlStrategyConstant.hpp"` in this file

Reply:
Hmm ... checking it out.

Reply:
@sconde So my mac and cee builds did not see this.  Also the autotester did not pick up on this either.  What are you building on?  (FYI, I do not think it is any problem adding this include.  Just interesting that only your build had an issue.)

Reply:
Building on my workstation. And I'm building from a clean build directory.

Reply:
Yeah, my builds to were from clean builds.  Hard to fix when you cannot reproduce the failures.  :(